### PR TITLE
GreatestCommonFactor: Use Math::BigInt & change to text template

### LINKED ISSUE #2985: GreatestCommonFactor: Use Math::BigInt
Instead of using a custom function, we should use `Math::BigInt::bgcd` to get the common divisors; additionally this will allow us to have an arbitrary amount of numbers (easily - in one go).

Remember to use `use Math::BigInt try => 'GMP';` for speed!

Also, if you're up for it, we need to update GreatestCommonFactor to use the new templates! The `text` template should be fine.

------
IA Page:  http://duck.co/ia/view/greatest_common_factor 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @austinheimark

### PR DESCRIPTION
**Pull request by rkokkelk**
Removed custom GCD function and replaced it with `Math::BigInt::bgcd`, and changed to `text` template for view

Fixes #2985

Instant Answer Page: https://duck.co/ia/view/greatest_common_factor

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @austinheimark

### CODE REVIEW

**Comment by daxtheduck** [comment]
@rkokkelk Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/GreatestCommonFactor.pm [hunk 1]
```diff
@@ -24,21 +25,19 @@ handle remainder => sub {
 
     my $result = shift @numbers;
     foreach (@numbers) {
-        $result = gcf($result, $_)
+        $result = Math::BigInt::bgcd($result, $_)
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/GreatestCommonFactor.pm]: We can just use `Math::BigInt::bgcd(@numbers)`.

File: t/GreatestCommonFactor.t [hunk 2]
```diff
@@ -13,121 +13,181 @@ ddg_goodie_test(
...
-            input     => ['4 and 6'],
-            operation => 'Greatest common factor',
-            result    => 2
+	  templates => {
+	    group => 'text'
+	  },
+	  data => {
+            subtitle => 'Greatest common factor: 4 and 6',
+            title    => 2
+          }
```
Comment (GuiltyDolphin) [inline t/GreatestCommonFactor.t]: Please use a function for generating the structured answers. See the [template](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/template/t/Example.t) for an example.

**Comment by GuiltyDolphin** [comment]
@rkokkelk Thanks! :dolphin:

**Pull request merged**